datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -366,6 +432,17 @@ pub fn date_bin(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     }
 }
 
+enum Interval {
+    Nanoseconds(i64),
+    Months(i64),
+}
+
+// Supported intervals:
+//  1. IntervalDayTime: this means that the stride is in days, hours, minutes, seconds and milliseconds
+//     We will assume month interval won't be converted into this type
+//     TODO (my next PR): for array data, the stride was converted into ScalarValue::IntervalDayTime somwhere
+//             for month interval. I need to find that and make it ScalarValue::IntervalMonthDayNano instead
```

Comment:
This is how to support month interval on array data. I think the change will be minimal but I need to figure out where